packages/backend/src/open_payments/receiver/service.test.ts

```diff
@@ -0,0 +1,217 @@
+          `${paymentPointer.url}/${CONNECTION_PATH}/${incomingPayment.connectionId}`
+        )
+
+        const clientGetConnectionSpy = jest
+          .spyOn(openPaymentsClient.ilpStreamConnection, 'get')
+          .mockImplementationOnce(async () =>
+            connectionService.get(incomingPayment).toOpenPaymentsType()
+          )
+
+        await expect(
+          receiverService.get(remoteUrl.href)
+        ).resolves.toMatchObject({
+          assetCode: paymentPointer.asset.code,
+          assetScale: paymentPointer.asset.scale,
+          incomingAmount: undefined,
+          receivedAmount: undefined,
+          ilpAddress: expect.any(String),
+          sharedSecret: expect.any(Buffer)
+        })
+        expect(clientGetConnectionSpy).toHaveBeenCalledWith({
+          url: remoteUrl.href
+        })
+      })
+
+      test('returns undefined for unknown connection', async (): Promise<void> => {
+        const paymentPointer = await createPaymentPointer(deps)
+
+        await expect(
+          receiverService.get(
+            `${paymentPointer.url}/${CONNECTION_PATH}/${uuid()}`
+          )
+        ).resolves.toBeUndefined()
+      })
+    })
+
+    describe('incoming payments', () => {
+      const INCOMING_PAYMENT_PATH = 'incoming-payments'
+
+      test('resolves local incoming payment', async () => {
+        const paymentPointer = await createPaymentPointer(deps, {
+          mockServerPort: Config.openPaymentsPort
+        })
+        const incomingPayment = await createIncomingPayment(deps, {
+          paymentPointerId: paymentPointer.id,
+          incomingAmount: {
+            value: BigInt(5),
+            assetCode: paymentPointer.asset.code,
+            assetScale: paymentPointer.asset.scale
+          }
+        })
+
+        await expect(
+          receiverService.get(incomingPayment.url)
+        ).resolves.toMatchObject({
+          assetCode: paymentPointer.asset.code,
+          assetScale: paymentPointer.asset.scale,
+          incomingAmount: incomingPayment.incomingAmount,
+          receivedAmount: incomingPayment.receivedAmount,
+          ilpAddress: expect.any(String),
+          sharedSecret: expect.any(Buffer)
+        })
+      })
+
+      test('resolves remote incoming payment', async () => {
+        const paymentPointer = await createPaymentPointer(deps)
+        const incomingPayment = await createIncomingPayment(deps, {
+          paymentPointerId: paymentPointer.id,
+          incomingAmount: {
+            value: BigInt(5),
+            assetCode: paymentPointer.asset.code,
+            assetScale: paymentPointer.asset.scale
+          }
+        })
+
+        const clientGetIncomingPaymentSpy = jest
+          .spyOn(openPaymentsClient.incomingPayment, 'get')
+          .mockImplementationOnce(async () =>
+            incomingPayment.toOpenPaymentsType({
+              ilpStreamConnection: connectionService.get(incomingPayment)
+            })
+          )
```

Comment:
Instead of using `nock` here, I just stub out the return value of the openPaymentsClient call